failsafe/failsafe.py

```diff
@@ -72,6 +74,10 @@ def __init__(self, retry_policy=None, circuit_breaker=None):
                 logger.debug("Circuit open, stopping execution")
                 raise CircuitOpen()
             try:
+                if self.retry_policy.backoff is not None:
+                    wait_for = self.retry_policy.backoff.for_attempt(context.attempts)
+                else:
+                    wait_for = None
```

Comment:
I am not a big fan of this `wait_for` variable here. It is only used 15 lines later and only used in some (rare) cases.

Instead of doing this, can we please return tuple `(should_retry, wait_for)` from `retry_policy.should_retry` function?

This function would handle this check: `if self.retry_policy.backoff is not None` and return `timedelta(0)` if `backoff` is not set.

Then we wound't need these 4 lines and it'd make this class nicer in general.